Mathlib/CategoryTheory/EffectiveEpi/Types.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2025 Yong-Gyu Choi. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yong-Gyu Choi
+-/
+import Mathlib.CategoryTheory.EffectiveEpi.RegularEpi
+import Mathlib.CategoryTheory.Limits.Types.Shapes
+
+/-!
+
+# Effective epimorphisms in the category of types
+
+We prove that effective epimorphisms in the category of types are precisely surjective functions.
+-/
+
+namespace CategoryTheory.Limits.Types
+
+universe u
+
+open Limits Function
+
+section CoequalizerPullbackCone
+
+variable {X Y : Type u} {f : X ⟶ Y}
+
+variable (f) in
+/-- The canonical function from the `Function.Coequalizer` of the two pullbackCone projections
+of `f : X ⟶ Y` along itself to `Y`. -/
+def fromCoequalizerPullbackCone :
+    Function.Coequalizer (Types.pullbackCone f f).fst (Types.pullbackCone f f).snd ⟶ Y :=
+  Function.Coequalizer.desc _ _ f (funext fun p => p.2)
+
+variable (f) in
+/-- `fromCoequalizerPullbackCone f` is always injective. -/
+lemma injective_fromCoequalizerPullbackCone : Injective (fromCoequalizerPullbackCone f) := by
+  intro z₁ z₂ h
+  rw [← Quot.out_eq z₁, ← Quot.out_eq z₂]
+  have h' : f (Quot.out z₁) = f (Quot.out z₂) := by
+    rwa [← Coequalizer.desc_mk _ _ f _ _, ← Coequalizer.desc_mk _ _ f _ _,
+      Coequalizer.mk, Coequalizer.mk, Quot.out_eq, Quot.out_eq]
+  exact Quot.sound (Coequalizer.Rel.intro (Subtype.mk (Prod.mk _ _) h'))
+
+/-- If `f : X ⟶ Y` is surjective, so is `fromCoequalizerPullbackCone f`. -/
+lemma surjective_fromCoequalizerPullbackCone (hf : Surjective f) :
+    Surjective (fromCoequalizerPullbackCone f) := by
+  intro y
+  obtain ⟨x, _⟩ := hf y
+  exact ⟨Quot.mk _ x, by simpa⟩
+
+/-- If `f : X ⟶ Y` is surjective, then `fromCoequalizerPullbackCone f` is surjective. -/
```

Comment:
Note that we do not require docstrings for lemmas, especially when the statement is very clear from the code, which is the case here:
```suggestion
/-- If `f : X ⟶ Y` is surjective, then `fromCoequalizerPullbackCone f` is bijective. -/
```